apps/ledger-live-mobile/src/screens/FirmwareUpdate/useUpdateFirmwareAndRestoreSettings.ts

```diff
@@ -0,0 +1,113 @@
+import staxLoadImage from "@ledgerhq/live-common/hw/staxLoadImage";
+import staxFetchImage from "@ledgerhq/live-common/hw/staxFetchImage";
+import { createAction as createStaxLoadImageAction } from "@ledgerhq/live-common/hw/actions/staxLoadImage";
+import { createAction as createStaxFetchImageAction } from "@ledgerhq/live-common/hw/actions/staxFetchImage";
+import { useUpdateFirmware } from "@ledgerhq/live-common/deviceSDK/hooks/useUpdateFirmware";
+import { Device, DeviceModelId } from "@ledgerhq/types-devices";
+import {
+  UpdateFirmwareActionState,
+  updateFirmwareActionArgs,
+} from "@ledgerhq/live-common/deviceSDK/actions/updateFirmware";
+import { Observable } from "rxjs";
+import { useEffect, useState } from "react";
+
+export type FirmwareUpdateParams = {
+  device: Device;
+  updateFirmwareAction?: (
+    args: updateFirmwareActionArgs,
+  ) => Observable<UpdateFirmwareActionState>;
+};
+
+export type UpdateStep =
+  | "start"
+  | "imageBackup"
+  | "firmwareUpdate"
+  | "languageRestore"
+  | "imageRestore"
+  | "appsRestore"
+  | "completed";
+
+const staxLoadImageAction = createStaxLoadImageAction(staxLoadImage);
+const staxFetchImageAction = createStaxFetchImageAction(staxFetchImage);
+
+export const useUpdateFirmwareAndRestoreSettings = ({
+  updateFirmwareAction,
+  device,
+}: FirmwareUpdateParams) => {
+  const [updateStep, setUpdateStep] = useState<UpdateStep>("start");
+
+  const staxFetchImageState = staxFetchImageAction.useHook(
+    updateStep === "imageBackup" ? device : null,
+    "",
```

Comment:
`Type '""' has no properties in common with type 'FetchImageRequest'.ts(2559)` TS not happy